lib/lz4hc.c

```diff
@@ -287,6 +302,39 @@ LZ4HC_InsertAndGetWiderMatch (
         }   }   }   }
     }  /* while ((matchIndex>=lowLimit) && (nbAttempts)) */
 
+    if (dict == usingDictCtx && nbAttempts && ip - base - lowLimit < MAX_DISTANCE) {
+        const ptrdiff_t dictIndexDelta = dictCtx->base - dictCtx->end + lowLimit;
+        /* bounds check, since we need to downcast */
```

Comment:
1 GB is unlikely to be large enough.

I suspect `dictCtx->base - dictCtx->end` could be correctly bounded (<= 64KB ?),
but `lowLimit` can be much larger, and becomes your real concern.

Reply:
(concept)
```
size_t const dictMaxOffset_st = dictCtx->end - dictCtx->base;
U32 const dictMaxOffset = (U32)dictMaxOffset_st;   // <--- assert();
U32 const dictIndexDelta = lowLimit - dictMaxOffset;   // defined
matchIndex = dictMatchIndex + dictIndexDelta;  // U32 = U32 + U32
```